File: from-0301-to-0400/0368-largest-divisible-subset/solution.py
class Solution(object):
    def largestDivisibleSubset(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        if not nums:
            return nums
        
        numss = sorted(nums)
        result = [[num] for num in numss]
        
        for i in range(len(numss)):
            for j in range(i):
                if numss[i] % numss[j] == 0 and len(result[i]) < len(result[j]) + 1:
                    result[i] = result[j] + [numss[i]]
                    
        return max(result, key=lambda x: len(x))

File: from-0301-to-0400/0368-largest-divisible-subset/solution.go
func largestDivisibleSubset(nums []int) []int {
    sort.Ints(nums)
    n := len(nums)
    result := make([][]int, n)
    for i := 0; i < n; i ++ {
        result[i] = []int{nums[i]}
    }

    for i := 0; i < n; i ++ {
        for j := 0; j < i; j ++ {
            if nums[i] % nums[j] == 0 && len(result[i]) < len(result[j]) + 1 {
                copy(result[i], result[j])
                result[i] = append(result[i], nums[i])
            }
        }
    }
    maxLen := -1
    ans := []int{}
    for i := 0; i < n; i ++ {
        if len(result[i]) > maxLen {
            maxLen = len(result[i])
            ans = result[i]
        }
    }
    return ans
}